salt/utils/thin.py

```diff
@@ -435,6 +436,7 @@ def get_tops(extra_mods="", so_mods=""):
     mods.append(contextvars)
     if has_immutables:
         mods.append(immutables)
+        mods.append(typing_extensions)
```

Comment:
I did some testing and I can only replicate the issue when my master and the target are python 3.6. If master is 3.8 and target is 3.6 it works. Can we add some logic here that only adds the typing_extensions if the master is running < python 3.8.